packages/catalog-readmodel-writer/test/consumerService.integration.test.ts

```diff
@@ -672,6 +696,892 @@ describe("database test", async () => {
+        interface: descriptorInterface,
+      };
+      const eservice: EService = {
+        ...mockEService,
+        descriptors: [draftDescriptor],
+      };
+      await writeInReadmodel(toReadModelEService(eservice), eservices, 1);
+
+      const updatedEService: EService = {
+        ...eservice,
+        descriptors: [
+          { ...draftDescriptor, serverUrls: [], interface: undefined },
+        ],
+      };
+      const payload: EServiceDescriptorInterfaceDeletedV2 = {
+        eservice: toEServiceV2(updatedEService),
+        descriptorId: draftDescriptor.id,
+        documentId: descriptorInterface.id,
+      };
+
+      const message: EServiceEventEnvelope = {
+        sequence_num: 1,
+        stream_id: mockEService.id,
+        version: 2,
+        type: "EServiceDescriptorInterfaceDeleted",
+        event_version: 2,
+        data: payload,
+      };
+      await handleMessageV2(message, eservices);
+
+      const retrievedEservice = await eservices.findOne({
+        "data.id": mockEService.id,
+      });
+
+      expect(retrievedEservice?.data).toEqual(
+        toReadModelEService(updatedEService)
+      );
+      expect(retrievedEservice?.metadata).toEqual({ version: 2 });
+    });
+
+    it("EServiceDescriptorDocumentDeleted", async () => {
+      const document = getMockDocument();
+      const draftDescriptor: Descriptor = {
+        ...getMockDescriptor(),
+        state: descriptorState.draft,
+        docs: [document],
+      };
+      const eservice: EService = {
+        ...mockEService,
+        descriptors: [draftDescriptor],
+      };
+      await writeInReadmodel(toReadModelEService(eservice), eservices, 1);
+
+      const updatedEService: EService = {
+        ...eservice,
+        descriptors: [{ ...draftDescriptor, docs: [] }],
+      };
+      const payload: EServiceDescriptorDocumentDeletedV2 = {
+        eservice: toEServiceV2(updatedEService),
+        descriptorId: draftDescriptor.id,
+        documentId: document.id,
+      };
+
+      const message: EServiceEventEnvelope = {
+        sequence_num: 1,
+        stream_id: mockEService.id,
+        version: 2,
+        type: "EServiceDescriptorDocumentDeleted",
+        event_version: 2,
+        data: payload,
+      };
+      await handleMessageV2(message, eservices);
+
+      const retrievedEservice = await eservices.findOne({
+        "data.id": mockEService.id,
+      });
+
+      expect(retrievedEservice?.data).toEqual(
+        toReadModelEService(updatedEService)
+      );
+      expect(retrievedEservice?.metadata).toEqual({ version: 2 });
+    });
+
+    it("EServiceRiskAnalysisAdded", async () => {
+      await writeInReadmodel<EServiceReadModel>(
+        toReadModelEService(mockEService),
+        eservices,
+        1
+      );
+
+      const mockRiskAnalysis = getMockValidRiskAnalysis("PA");
+      const updatedEService: EService = {
+        ...mockEService,
+        riskAnalysis: [...mockEService.riskAnalysis, mockRiskAnalysis],
```

Comment:
```suggestion
        riskAnalysis: [mockRiskAnalysis],
```
This is enough since `mockEService.riskAnalysis` is empty

Reply:
Updated